packages/pds/tests/account-migration.test.ts

```diff
@@ -0,0 +1,215 @@
+      validDid: false,
+      repoBlocks: 2, // commit & empty data root
+      indexedRecords: 0,
+      privateStateValues: 0,
+      expectedBlobs: 0,
+      importedBlobs: 0,
+    })
+
+    const repoRes = await oldAgent.com.atproto.sync.getRepo({ did: alice })
+    const carBlocks = await readCar(repoRes.data)
+
+    await newAgent.com.atproto.repo.importRepo(repoRes.data, {
+      encoding: 'application/vnd.ipld.car',
+    })
+
+    const statusRes2 = await newAgent.com.atproto.server.checkAccountStatus()
+    expect(statusRes2.data).toMatchObject({
+      activated: false,
+      validDid: false,
+      indexedRecords: 103,
+      privateStateValues: 0,
+      expectedBlobs: 3,
+      importedBlobs: 0,
+    })
+    expect(statusRes2.data.repoBlocks).toBe(carBlocks.blocks.size)
+
+    const missingBlobs = await newAgent.com.atproto.repo.listMissingBlobs()
+    expect(missingBlobs.data.blobs.length).toBe(3)
+
+    let blobCursor: string | undefined = undefined
+    do {
+      const listedBlobs = await oldAgent.com.atproto.sync.listBlobs({
+        did: alice,
+        cursor: blobCursor,
+      })
+      for (const cid of listedBlobs.data.cids) {
+        const blobRes = await oldAgent.com.atproto.sync.getBlob({
+          did: alice,
+          cid,
+        })
+        await newAgent.com.atproto.repo.uploadBlob(blobRes.data, {
+          encoding: blobRes.headers['content-type'],
+        })
+      }
+      blobCursor = listedBlobs.data.cursor
+    } while (blobCursor)
+
+    const statusRes3 = await newAgent.com.atproto.server.checkAccountStatus()
+    expect(statusRes3.data.expectedBlobs).toBe(3)
+    expect(statusRes3.data.importedBlobs).toBe(3)
+
+    const prefs = await oldAgent.api.app.bsky.actor.getPreferences()
+    await newAgent.api.app.bsky.actor.putPreferences(prefs.data)
+
+    const getDidCredentials =
+      await newAgent.com.atproto.identity.getRecommendedDidCredentials()
+
+    await oldAgent.com.atproto.identity.requestPlcOperationSignature()
+    const res = await network.pds.ctx.accountManager.db.db
+      .selectFrom('email_token')
+      .selectAll()
+      .where('did', '=', alice)
+      .where('purpose', '=', 'plc_operation')
+      .executeTakeFirst()
+    const token = res?.token
+    assert(token)
+
+    const plcOp = await oldAgent.com.atproto.identity.signPlcOperation({
+      token,
+      ...getDidCredentials.data,
+    })
+
+    await newAgent.com.atproto.identity.submitPlcOperation({
+      operation: plcOp.data.operation,
+    })
+
+    await newAgent.com.atproto.server.activateAccount()
+
+    const statusRes4 = await newAgent.com.atproto.server.checkAccountStatus()
+    expect(statusRes4.data).toMatchObject({
+      activated: true,
+      validDid: true,
+      indexedRecords: 103,
+      privateStateValues: 0,
+      expectedBlobs: 3,
+      importedBlobs: 3,
+    })
+
+    await oldAgent.com.atproto.server.deactivateAccount({
+      deleteAfter: new Date(Date.now() + HOUR).toISOString(),
+    })
```

Comment:
What does this part test—that you can write even after marked for deactivation?

Reply:
I really just added it in order to finish documentation of the flow

I'll add an additional test to just check the status on the old pds 👌